libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -78,277 +78,305 @@ AP_GPS *AP_GPS::_singleton;
     // @User: Advanced
-    AP_GROUPINFO("DRV_OPTIONS", 22, AP_GPS, _driver_options, 0),
+    AP_GROUPINFO("_DRV_OPTIONS", 22, AP_GPS, _driver_options, 0),
 #endif
 
-    // @Param: COM_PORT
+    // @Param: _COM_PORT
     // @DisplayName: GPS physical COM port
     // @Description: The physical COM port on the connected device, currently only applies to SBF GPS
     // @Range: 0 10
     // @Increment: 1
     // @User: Advanced
     // @RebootRequired: True
-    AP_GROUPINFO("COM_PORT", 23, AP_GPS, _com_port[0], HAL_GPS_COM_PORT_DEFAULT),
+    AP_GROUPINFO("_COM_PORT", 23, AP_GPS, _com_port[0], HAL_GPS_COM_PORT_DEFAULT),
 
 #if GPS_MAX_RECEIVERS > 1
-    // @Param: COM_PORT2
+    // @Param: _COM_PORT2
     // @DisplayName: GPS physical COM port
     // @Description: The physical COM port on the connected device, currently only applies to SBF GPS
     // @Range: 0 10
     // @Increment: 1
     // @User: Advanced
     // @RebootRequired: True
-    AP_GROUPINFO("COM_PORT2", 24, AP_GPS, _com_port[1], HAL_GPS_COM_PORT_DEFAULT),
+    AP_GROUPINFO("_COM_PORT2", 24, AP_GPS, _com_port[1], HAL_GPS_COM_PORT_DEFAULT),
 #endif
 
 #if GPS_MOVING_BASELINE
 
-    // @Group: MB1_
+    // @Group: _MB1_
     // @Path: MovingBase.cpp
-    AP_SUBGROUPINFO(mb_params[0], "MB1_", 25, AP_GPS, MovingBase),
+    AP_SUBGROUPINFO(mb_params[0], "_MB1_", 25, AP_GPS, MovingBase),
 
 #if GPS_MAX_RECEIVERS > 1
-    // @Group: MB2_
+    // @Group: _MB2_
     // @Path: MovingBase.cpp
-    AP_SUBGROUPINFO(mb_params[1], "MB2_", 26, AP_GPS, MovingBase),
+    AP_SUBGROUPINFO(mb_params[1], "_MB2_", 26, AP_GPS, MovingBase),
 #endif // GPS_MAX_RECEIVERS > 1
 
 #endif // GPS_MOVING_BASELINE
 
 #if GPS_MAX_RECEIVERS > 1
-    // @Param: PRIMARY
+    // @Param: _PRIMARY
     // @DisplayName: Primary GPS
     // @Description: This GPS will be used when GPS_AUTO_SWITCH is 0 and used preferentially with GPS_AUTO_SWITCH = 4.
     // @Increment: 1
     // @Values: 0:FirstGPS, 1:SecondGPS
     // @User: Advanced
-    AP_GROUPINFO("PRIMARY", 27, AP_GPS, _primary, 0),
+    AP_GROUPINFO("_PRIMARY", 27, AP_GPS, _primary, 0),
+#endif
+
+#if GPS_MAX_RECEIVERS > 1 && HAL_ENABLE_LIBUAVCAN_DRIVERS
+    // @Param: _CAN_NODEID1
+    // @DisplayName: GPS Node ID 1
+    // @Description: GPS Node id for discovered first.
+    // @ReadOnly: True
+    // @User: Advanced
+    AP_GROUPINFO("_CAN_NODEID1", 28, AP_GPS, _node_id[0], 0),
+
+    // @Param: _CAN_NODEID2
+    // @DisplayName: GPS Node ID 2
+    // @Description: GPS Node id for discovered second.
+    // @ReadOnly: True
+    // @User: Advanced
+    AP_GROUPINFO("_CAN_NODEID2", 29, AP_GPS, _node_id[1], 0),
+
+    // @Param: 1_CAN_OVRIDE
+    // @DisplayName: First UAVCAN GPS NODE ID
+    // @Description: GPS Node id for first GPS. If 0 the gps will be automatically selected on first come basis.
+    // @User: Advanced
+    AP_GROUPINFO("1_CAN_OVRIDE", 30, AP_GPS, _override_node_id[0], 0),
```

Comment:
This seems like a regression in parameter naming, as it will break the tree in most GCS's, and logically ungroups it in the wiki. At which point we can undo all the other `_` prefix changes...
```suggestion
    AP_GROUPINFO("_CAN_1_ID", 30, AP_GPS, _override_node_id[0], 0),
```

Reply:
This ID is not related to GPS instance, it's just a list of detected CAN GPS IDs not attached to particular instance. So we actually don't want these grouped under instance, but as common parameters.